qemu/tests/dump_guest_memory.py

```diff
@@ -0,0 +1,229 @@
+
+    def execute_qmp_cmd(qmp_cmd, expect_result):
+        """
+        Execute qmp command and check if result as expect
+
+        :param qmp_cmd: qmp command
+        :type qmp_cmd: str
+        :param expect_result: expect result of qmp command
+        :type expect_result: str
+        """
+        # qmp command
+        try:
+            # Testing command
+            logging.info("Run qmp command '%s'.", qmp_cmd)
+            qmp_o = qmp_port.send_args_cmd(qmp_cmd)
+            if not isinstance(qmp_o, list):
+                qmp_o = [qmp_o]
+            logging.debug("QMP command: '%s' \n Output: '%s'", qmp_cmd, qmp_o)
+        except Exception as err:
+            qmp_o = err.data
+            logging.info(err)
+
+        if result_check == "equal" or result_check == "contain":
+            logging.info("Verify qmp command '%s'." % qmp_cmd)
+            check_result(qmp_o, eval(expect_result))
+
+    def check_dump_file(file_name):
+        """
+        Use crash to check dump file
+
+        :param file_name: dump file name
+        :type file_name: str
+
+        """
+        try:
+            kernel_version = process.getoutput("uname -r")
+            crash_script = "/home/crash.cmd"
+            process.run("echo bt > %s" % crash_script)
+            process.run("echo quit >> %s" % crash_script)
+            crash_cmd = "crash -i %s /usr/lib/debug/lib/modules/%s/vmlinux "
+            crash_cmd %= (crash_script, kernel_version)
+            crash_cmd += file_name
+            output = process.getoutput(crash_cmd)
+            logging.debug("%s", output)
+        except process.CmdError as err:
+            output = err.result.stderr.decode()
+        if output.find("error") >= 0:
+            test.fail("vmcore corrupt")
+
+    # install crash/gdb
+    install_package("crash")
+    install_package("gdb")
+
+    qemu_binary = utils_misc.get_qemu_binary(params)
+    if not utils_misc.qemu_has_option("qmp", qemu_binary):
+        test.cancel("Host qemu does not support qmp.")
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+
+    session = vm.wait_for_login(timeout=int(params.get("login_timeout", 360)))
+
+    qmp_ports = vm.get_monitors_by_type('qmp')
+    if qmp_ports:
+        qmp_port = qmp_ports[0]
+    else:
+        test.error("Incorrect configuration, no QMP monitor found.")
+
+    # install kernel-debuginfo and kernel-debuginfo-common in host
+    process.run("yum -y install --enablerepo='*debuginfo' kernel-debuginfo*")
+
+    qmp_cmd = params.get("qmp_cmd")
+    query_qmp_cmd = params.get("query_qmp_cmd")
+    result_check = params.get("cmd_result_check")
+    query_cmd_return_value = params.get("query_cmd_return_value")
+    strict_match = params.get("strict_match", "yes") == 'yes'
+    expect_result = params.get("cmd_return_value", "[]")
+
+    # execute dump qmp command
+    execute_qmp_cmd(qmp_cmd, expect_result)
+
+    if "paging=false" in qmp_cmd:
+        time.sleep(10)
```

Comment:
What this sleep for? Any other precise checking for what you are waiting for beside sleep?